### ISSUE TITLE
Migrate `nitrokey-firmware`

### ISSUE THREAD

**Issue by augustebaum** [ngi-nix/ngipkgs#70]
https://github.com/ngi-nix/nitrokey-firmware

Tracking issue: https://github.com/ngi-nix/ngipkgs/issues/12.

**Comment by mightyiam** [comment]
@fricklerhandwerk what NGI program is this in please?

**Comment by wamirez** [comment]
Closing issue. Superseded by:

- https://github.com/ngi-nix/ngipkgs/pull/90

**Comment by lorenzleutgeb** [comment]
I believe #90 stalled. Maybe @mightyiam knows more?

**Comment by lorenzleutgeb** [comment]
@mightyiam I guess that it would be helpful for @jleightcap to know *why* it stopped and *which* branches to look at 😺

**Comment by mightyiam** [comment]
Not for technical reasons. The branches probably have not unrelated names. Sorry, that's all I got without looking through the contents.

**Comment by mightyiam** [comment]
Don't miss out on mob/* branches. They may have good progress.

**Comment by fricklerhandwerk** [comment]
@jleightcap yes that issues is free for grabs, would be cool if we had a end-to-end Nitrokey setup. There could be enough work for an entire Summer of Nix team.

**Comment by RCoeurjoly** [comment]
There are two packages that are failing for the same reason, that we are trying to debug: nitrokey-fido2, nitrokey-pro. 
   
    We are working on this branch: https://github.com/jleightcap/nitrokey-firmware/tree/fern/2024-08-13
steps to reproduce: `nix build .#nitrokey-fido2`
    We get the following error

>          error: attribute 'version' missing
> 
>        at /nix/store/h60m1fwahjd2mv6gsg77ji3vb4gpj4dk-source/pkgs/os-specific/linux/systemd/default.nix:119:59:
> 
>           118|     # assumes hard floats
>           119|     && (stdenv.hostPlatform.isAarch -> lib.versionAtLeast stdenv.hostPlatform.parsed.cpu.version "6")
>              |                                                           ^
>           120|     # see https://github.com/NixOS/nixpkgs/pull/194149#issuecomment-1266642211
> 

We don't in the REPL that cpu has an attribute called version.
> nix-repl> :p outputs.packages.x86_64-linux.nitrokey-fido2.stdenv.hostPlatform.parsed.cpu
> { _type = "cpu-type"; bits = 32; family = "arm"; name = "arm"; significantByte = { _type = "significant-byte"; name = "littleEndian"; }; }

**Comment by fricklerhandwerk** [comment]
Closed by:
- https://github.com/NixOS/nixpkgs/pull/337966
- https://github.com/NixOS/nixpkgs/pull/337959
- https://github.com/NixOS/nixpkgs/pull/337956
- https://github.com/NixOS/nixpkgs/pull/337954
- https://github.com/NixOS/nixpkgs/pull/337951

Awesome work @kiike @amerinor01 @jleightcap @imadnyc @RCoeurjoly @wegank!

**Comment by robinkrahl** [comment]
Great to see that these have been integrated into Nixpkgs!  The `nitrokey-firmware` repository also included a [derivation for the Nitrokey 3 firmware](https://github.com/ngi-nix/nitrokey-firmware/blob/master/devices/nitrokey-3.nix).  AFAIS this has not been included in NGIpkgs or Nixpkgs.  Is that intentional?

**Comment by fricklerhandwerk** [comment]
Good question! @kiike @jleightcap is there a particular reason?

**Comment by kiike** [comment]
Yes. In short, we found a bug in the Rust bootstrapping process that prevents building for the particular platform that nitro key 3 is using. The initial approach I took worked around the issue but needed changing the way Rust in Nix is built and apparently the Nixpkgs rust team are not interested in changing this, as the change is quite huge.So, to unblock this I investigated the bug and posted the issue and some potential approaches to solve it some weeks ago. Eventually the fix was merged last week. Now I’m monitoring the release progress to adopt the change in nixpkgs, which will consist of adding quotes around the platform names.bootstrap/configure.py: targets with a dot in the name prevent finding the correct target section in `config.toml.example` · Issue #130602 · rust-lang/rustgithub.comhttps://github.com/NixOS/nixpkgs/pull/342627 Message ID: ***@***.***>
